### PR TITLE
Fix recently-pushed-branches once and for all

### PR DESCRIPTION
**Pull request by bfred-it**
Fixes https://github.com/sindresorhus/refined-github/issues/1109

- Repositions notification to black header bar
- Adds notification on all pages of repo (but it only ever "pops in" once)
- Makes the whole feature disableable (so if people want the regular full-width bar, they can)

### CODE REVIEW

File: source/features/recently-pushed-branches-enhancements.tsx [hunk 1]
```diff
@@ -13,30 +13,30 @@ import {getRepoURL} from '../libs/utils';
...
-	const response = await fetch(codeTabURL);
-	const html = await response.text();
-
-	// https://github.com/sindresorhus/refined-github/issues/216
-	if (html.includes(fragmentURL)) {
-		select('.repository-content').prepend(<include-fragment src={fragmentURL}></include-fragment>);
-	}
+
+	return (
+		select(`[data-url='${fragmentURL}'], [src='${fragmentURL}']`) || // Already on page
```
Comment (notlmn) [inline source/features/recently-pushed-branches-enhancements.tsx]: Explicit `if-else` using `select.exists()` would be a better choice here.

Took me a minute to understand what was happening, with all the combined logical OR and inline comments.

**Comment by notlmn** [comment]
How about we also change the background colors to make it a bit less distracting?

File: source/features/recently-pushed-branches-enhancements.tsx (hunk 1 above)
Comment (bfred-it) [reply source/features/recently-pushed-branches-enhancements.tsx]: Indeed. Should be better now

**Comment by bfred-it** [comment]
color

File: source/content.css [hunk 2]
```diff
@@ -745,63 +745,70 @@ a.tabnav-extra[href$='mastering-markdown/'] {
...
 }
-.RecentBranches {
+.rgh-recently-pushed-branches .RecentBranches {
 	display: flex;
 	width: max-content;
-	margin-top: 10px; /* Add extra padding to make it look right on forked repos */
 	margin-left: auto;
 	flex-direction: column-reverse;
+	background: none;
+	border-color: #fff;
```
Comment (notlmn) [inline source/content.css]:
```suggestion
	border-color: currentColor;
```

**Comment by notlmn** [comment]
Is the time any helpful here, just thinking 🤔?

**Comment by bfred-it** [comment]
It helps you understand if it's something that was pushed just now or an old push. Not _really_ useful but it's not a huge distraction.

**Comment by bfred-it** [comment]
@sindresorhus if the style looks ok I can finalize this

**Comment by sindresorhus** [comment]
I found the button very useful. I always use it to create a new PR.

The style in https://github.com/sindresorhus/refined-github/pull/1909#issuecomment-479584096 looks very good. I would make the gradient gray button slightly darker.

**Comment by sindresorhus** [comment]
Can you rebase from master?

**Comment by bfred-it** [comment]
I don't know how dark you're envisioning it. Mind playing around with the filters in 3f0c55e?

Current look with multiple branches hidden

Screenshot 2019-04-18 at 19 18 09

**Comment by sindresorhus** [comment]
Done

**Comment by notlmn** [comment]
@bfred-it Still open to some changes?

### Visual changes

| Collapsed | Expanded |
|:-:|:-:|
| image | image |

### Source diff

``` diff
diff --git a/source/content.css b/source/content.css
index 5378d509e3b3..1d65ff42f8b8 100644
--- a/source/content.css
        text-indent: -999999px; /* Hide text */
-}
-.rgh-recently-pushed-branches .RecentBranches:not(:hover) .btn {
-       filter: grayscale(1) brightness(0.8) contrast(2);
+       background: #34d058;
 }
 .rgh-recently-pushed-branches .RecentBranches .btn + div {
        margin-right: 8px;
-       float: right;
        font-size: 11px;
 }
 .rgh-recently-pushed-branches .RecentBranches .btn .octicon {
@@ -797,9 +794,14 @@ a.tabnav-extra[href$='mastering-markdown/'] {
        border-bottom: 1px solid;
 }
 .rgh-recently-pushed-branches .RecentBranches-item {
+       display: flex;
+       flex-direction: row-reverse;
        color: inherit;
        padding: 4px !important;
 }
+.rgh-recently-pushed-branches .RecentBranches-item:not(:hover) .btn {
+       filter: grayscale(1) brightness(0.8) contrast(2);
+}
 .rgh-recently-pushed-branches .RecentBranches:not(:hover) .RecentBranches-item:not(:last-child) {
        /* Hide older branches */
        display: none;
```

### Changes

- Reduce `border-radius` in favor for non-retina display
- Change gradient background of button to solid background
- Use `flex`, drop `float`
- Move hover from items to individual item

**Comment by bfred-it** [comment]
I'm ok with with making the gray flat to make it fit better up there, but not the green, as we still need to match GitHub's style.

**Comment by bfred-it** [comment]
I applied the flat color on `:not(:hover)`, but:

- `float: right` is just one easy property
- The only reason we grayscale the button is to make it fit in the header, but once the user is "onto it", the whole widget should be enabled

**Comment by busches** [comment]
This is broke on GHE of course :) I'll take a look at it tomorrow.

**Pull request merged**